test/onigumo_cli_test.exs

```diff
@@ -23,18 +33,30 @@ defmodule OnigumoCLITest do
       assert usage_message_printed?(fn -> Onigumo.CLI.main(["Downloader", "Parser"]) end)
     end
 
-    test("run CLI with invalid switch") do
-      assert usage_message_printed?(fn -> Onigumo.CLI.main(["--invalid"]) end)
+    for switch <- @invalid_switches do
+      test("run CLI with invalid switch #{inspect(switch)}") do
+        assert usage_message_printed?(fn -> Onigumo.CLI.main([unquote(switch)]) end)
+      end
     end
 
     @tag :tmp_dir
     test("run CLI with 'downloader' argument passing cwd", %{tmp_dir: tmp_dir}) do
-      expect(OnigumoDownloaderMock, :main, fn root_path -> root_path end)
+      expect(OnigumoDownloaderMock, :main, fn working_dir -> working_dir end)
 
       File.cd(tmp_dir)
       assert Onigumo.CLI.main(["downloader"]) == tmp_dir
     end
 
+
+    for switch <- @working_dir_switches do
+      @tag :tmp_dir
+      test("run CLI 'downloader' with #{inspect(switch)} switch", %{tmp_dir: tmp_dir}) do
+        expect(OnigumoDownloaderMock, :main, fn working_dir -> working_dir end)
+
+        assert Onigumo.CLI.main(["downloader", unquote(switch), tmp_dir]) == tmp_dir
```

Comment:
It’s nice what an ingenious trick we found to check the actual value of the working dir. Maybe it’s time to create a separate function to parse the CLI options. Not in this PR, though.